packages/react-button/src/components/Button/useButton.ts

```diff
@@ -1,28 +1,34 @@
 import * as React from 'react';
-import { getStyleFromPropsAndOptions } from '@fluentui/react-theme-provider';
-import { useFocusRects } from '@uifabric/utilities';
-import { ComposePreparedOptions } from '@fluentui/react-compose';
-import { ButtonProps, ButtonState } from './Button.types';
+import { ButtonState } from './Button.types';
+// import { useMergedRefs } from '@uifabric/react-hooks';
 
 /**
- * The useButton hook processes the Button component props and returns state.
- * @param props - Button props to derive state from.
+ * The useButton hook processes the Button draft state.
+ * @param draftState - Button draft state to mutate.
  */
-export const useButton = (
-  props: ButtonProps,
-  ref: React.Ref<HTMLElement>,
-  options: ComposePreparedOptions,
-): ButtonState => {
-  const buttonRef = React.useRef<HTMLButtonElement | null>(null);
+export const useButton = (draftState: ButtonState) => {
+  // Update the button's tab-index, keyboard handling, and aria attributes.
+  if (draftState.as !== 'button') {
```

Comment:
nit: can we invert these if conditions so we can return early without nesting code for readability